In [55]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from sklearn.tree import export_graphviz,plot_tree
from sklearn import tree

In [56]:
df=pd.read_csv("kyphosis.csv")

In [57]:
df

Kyphosis  Age  Number  Start
0    absent   71       3      5
1    absent  158       3     14
2   present  128       4      5
3    absent    2       5      1
4    absent    1       4     15
..      ...  ...     ...    ...
76  present  157       3     13
77   absent   26       7     13
78   absent  120       2     13
79  present   42       7      6
80   absent   36       4     13

[81 rows x 4 columns]

In [58]:
sns.pairplot(df)

In [59]:
from sklearn.model_selection import train_test_split

In [60]:
from sklearn.tree import DecisionTreeClassifier

In [61]:
from sklearn.ensemble import RandomForestClassifier

In [62]:
from sklearn.metrics import classification_report,confusion_matrix

In [63]:
df.columns

Index(['Kyphosis', 'Age', 'Number', 'Start'], dtype='object')

In [64]:
X=df[ ['Age', 'Number', 'Start']]

In [65]:
y=df['Kyphosis']

In [66]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=101)

In [67]:
dt=DecisionTreeClassifier()

In [68]:
dt.fit(X_train,y_train)

DecisionTreeClassifier()

In [69]:
pred=dt.predict(X_test)

# report based on Decision tree

In [70]:
print(classification_report(y_test,pred))
print('\n')
print(confusion_matrix(y_test,pred))

              precision    recall  f1-score   support

      absent       0.71      0.79      0.75        19
     present       0.33      0.25      0.29         8

    accuracy                           0.63        27
   macro avg       0.52      0.52      0.52        27
weighted avg       0.60      0.63      0.61        27



[[15  4]
 [ 6  2]]


..................................................................................

In [72]:
rf=RandomForestClassifier(n_estimators=100)

In [73]:
rf.fit(X_train,y_train)

RandomForestClassifier()

In [74]:
pred=rf.predict(X_test)

# report based on random forest

In [20]:
print(classification_report(y_test,pred))
print('\n')
print(confusion_matrix(y_test,pred))

              precision    recall  f1-score   support

      absent       0.76      1.00      0.86        19
     present       1.00      0.25      0.40         8

    accuracy                           0.78        27
   macro avg       0.88      0.62      0.63        27
weighted avg       0.83      0.78      0.73        27



[[19  0]
 [ 6  2]]


In this model the accuracy of random forest is greater than decision tree which 78%...

.........................................................................